Cython/Compiler/Code.py

```diff
@@ -1207,7 +1209,8 @@ def initialize_main_c_code(self):
         w = self.parts['cached_constants']
         w.enter_cfunc_scope()
         w.putln("")
-        w.putln("static CYTHON_SMALL_CODE int __Pyx_InitCachedConstants(void) {")
+        w.putln("static CYTHON_SMALL_CODE int __Pyx_InitCachedConstants(%s *%s) {" % (
+            Naming.modulestate_cname, Naming.modulestate_cname))
```

Comment:
Using the same name for the type and the argument here seems … funny. You probably did that to avoid renaming the usages, right? It still looks weird, more like a typo at first sight.

Reply:
Yes. Right now `Naming.modulestate_cname` is used for (at least) three different things: a type, a "getter"-functionlike macro, and a value. I'll give all three uses separate names and hopefully it'll be a bit clearer.